docroot/themes/humsci/humsci_traditional/humsci_traditional.theme

```diff
@@ -30,6 +30,13 @@ function humsci_traditional_preprocess_html(&$vars) {
     // Set html_attributes for html DOM element
     $vars['html_attributes']->addClass('hb-dark-inversion');
   }
+
+  // Mega Menu hides standard menu on desktop setting.
+  $use_megamenu = theme_get_setting('megamenu_toggle');
+  if ($use_megamenu) {
+    // Load base megamenu CSS when megamenu is enabled
+    $vars['#attached']['library'][] =  'humsci_traditional/megamenu';
```

Comment:
This should be added at the block layer using a `preprocess_block__we_megamenu` I believe. Then you don't need to have the condition if the theme has it turned on

Reply:
Please take a look at the latest commit. I've changed it to we_megamenu_preprocess_block and tested Traditional and Colorful with the mega menu disabled and enabled. Appears to load the base CSS library only when it's enabled as hoped.

Reply:
Now this function will never run. A hook does not begin like this. Target the specific megemu block and rove the theme get settings line